### PR TITLE
Storable: modernization and clean up

### PR DESCRIPTION
**Pull request by haarg**
This cleans up and modernized various parts of Storable.

  * various files are moved around to better match modern practices
  * strict is enabled in the module and tests
  * warnings are enabled in the tests
  * whitespace is normalized at 4 spaces
  * removed various minor bits of code and comments that have been irrelevant for decades

### CODE REVIEW

File: dist/Storable/MANIFEST.SKIP [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+\bcover_db\b
+\bcovered\b
+
+\B\.prove$
+
+^MYMETA\.
+^META_new\.(?:json|yml)
+
+^Storable-[0-9.]+
+^Storable-[0-9.]+
```
Comment (tonycoz) [inline dist/Storable/MANIFEST.SKIP]: duplicate

File: dist/Storable/t/recurse.t [hunk 2]
```diff
@@ -7,14 +7,6 @@
 #
 use Config;
 
-sub BEGIN {
-    unshift @INC, 't';
-    if ($ENV{PERL_CORE} and $Config{'extensions'} !~ /\bStorable\b/) {
```
Comment (tonycoz) [inline dist/Storable/t/recurse.t:13]: You've removed the last use of %Config, but not the `use Config;`

**Comment by jkeenan** [comment]
Could these 22 commits be squashed into a much smaller number before we merge this to blead?

I ask so that we can reduce the number of commits which have to be traversed during future bisections.

Also:  the Changelog will at some point need a line for the latest version and release timestamp.

Thanks for undertaking this cleanup.

**Review by tonycoz — APPROVED** [review]
I think the separate commits are fine, they're not going to noticeably increase the length of a bisect.

I do think it should be merged as described in [perlgit](https://github.com/Perl/perl5/blob/9a99c2bec339a81cc62c0d6a25c4e1343be0eaad/pod/perlgit.pod?plain=1#L750) to group all the commits together (let me know if you'd like me to do it)

**Comment by tonycoz** [comment]
It looks like 5528f95b and only that commit from this PR has been applied to blead as bd8229e704.

**Comment by haarg** [comment]
> It looks like [5528f95](https://github.com/Perl/perl5/commit/5528f95bed1ffd316ee778d8e1fd8e68d0600766) and only that commit from this PR has been applied to blead as [bd8229e](https://github.com/Perl/perl5/commit/bd8229e7049add6f6da46c15d55a498e9c298b76).

I considered that commit just a simple fix, so I pulled it out of this PR.

I also reformed some of the commits, combining a few of them, and making sure they all pass the tests individually.

I can merge this with a merge commit it still looks fine.

**Comment by tonycoz** [comment]
This still looks fine to me.

**Pull request merged**